modules/generic/src/main/scala/GenericAlgebra.scala

```diff
@@ -0,0 +1,76 @@
+package scalaz
+
+package schema
+
+package generic
+
+trait GenericSchemaModule[R <: Realisation] extends SchemaModule[R] {
+
+  import Schema._
+  //import R._
+
+  type Id[A] = A
+
+  def covariantTargetFunctor[H[_]](
+    primNT: R.Prim ~> H,
+    seqNT: H ~> λ[X => H[List[X]]],
+    prodLabelNT: λ[X => (R.ProductTermId, H[X])] ~> H,
```

Comment:
I suppose that if we rearrange the type parameters of `ProductTerm` so that `A` comes last, this NT could be be expressed (arguably) more simply as `ProductTerm[R.Prim, R.SumTermId, R.ProductTermId, H, ?] ~> H`.

Moreover, we could define useful type aliases in `SchemaModule` in the form of:
```scala
type RProductTerm[F[_], A] = ProductTerm[R.Prim, R.SumTermId, R.ProductTermId, F, A]
```

and further simplify our type signatures.

Reply:
Also, maybe we should bundle these parameters into a single datatype.

Reply:
introduced `RProductTerm` and `RSumTerm` on `SchemaModule` and simplified the NTs accordingly. worked out nicely